e2e/filesystem-upload.test.js

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const Path = require('path');
+const should = require('should');
+
+const {
+    importFile,
+    getTargetFolder,
+    getTestOptions,
+    setCredentials,
+    doesAemPathExist,
+    deleteAemPath,
+    getHttpClient
+} = require('./e2eutils');
+
+const DirectBinaryUploadOptions = importFile('direct-binary-upload-options');
+const FileSystemUploadOptions = importFile('filesystem-upload-options');
+const FileSystemUpload = importFile('filesystem-upload');
+
+describe('FileSystemUpload end-to-end tests', function() {
```

Comment:
That's a great idea. Do you know of a way to corrupt a file such that it will throw errors when it's read?

Reply:
Some ways:
- (Manual) Open any JPEG image in a text editor (visual studio is fine) and randomly remove characters (rather at the beginning of the file).
- Open any JPEG image in a text editor (visual studio is fine) and randomly add characters (rather at the beginning of the file). Usually I add something like "This is a test", so if I reopen the file I see it's voluntarily corrupt for testing.
- (Automated) (NodeJS) Open the file, start reading it, stop randomly while reading (between one third and half the size)  and drop the read content into a new file. The resulting file will most likely be corrupt because it's truncated. This may not always work if you truncate only at the end.
- (Automated) Open the file, start reading it, randomly add text into the stream (like `this is a test`), one or more times. Drop the stream content in a new file, which should then be flagged as corrupt.